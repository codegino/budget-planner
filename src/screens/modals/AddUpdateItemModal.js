import React from 'react';
import { connect } from 'react-redux';
import { View, Text, TouchableNativeFeedback } from 'react-native';
import dateformat from 'dateformat';
import _ from 'lodash';

import DefaultInput from '../../components/input/DefaultInput';
import DefaultDatePicker from '../../components/datepicker/DefaultDatepicker';
import DefaultButton from '../../components/button/DefaultButton';
import { addExpenses, updateExpenses } from '../../store/actions/expenses';
import styles from './addUpdateItemStyle';
import DefaultPicker from '../../components/picker/DefaultPicker';
import { updateBudgetPlan } from '../../store/actions/budgetPlan';

type Props = {
  addExpenses: Function,
  updateExpenses: Function,
  item: {
    name: string,
    price: number,
    date: string,
    id: string,
    category: string,
  },
  categories: Function,
  navigator: Object,
  mode: string,
  getBudgetPlan: Function,
  updateBudgetPlan: Function,
};

class AddNewItemModal extends React.Component<Props> {
  constructor(props) {
    super(props);
    if (props.mode === 'edit') {
      this.state = {
        name: props.item.name,
        price: props.item.price.toString(),
        date: props.item.date,
        mode: props.mode,
        category: props.item.category,
      };
    } else {
      const categories = this.props.categories(this.state.date.substring(3, 11));
      this.state.category = categories.length > 0 ? categories[0] : '';
    }
  }

  state = {
    name: '',
    price: '',
    date: dateformat(new Date(), 'dd-mmm-yyyy'),
    category: '',
    mode: 'add',
  }

  onNameChangeHandler = (value) => {
    this.setState(prevState => ({
      ...prevState,
      name: value,
    }));
  };

  onPriceChangeHandler = (value) => {
    this.setState(prevState => ({
      ...prevState,
      price: value,
    }));
  };

  onDateChangeHandler = (value) => {
    const categories = this.props.categories(value.substring(3, 11));
    this.setState(prevState => ({
      ...prevState,
      date: value,
      category: categories.length > 0 ? categories[0] : '',
    }));
  };

  onCategoryChangeHandler = (value) => {
    this.setState(prevState => ({
      ...prevState,
      category: value,
    }));
  };

  onAddItemHandler = () => {
    if (this.validate() === false) {
      return;
    }

    const item = {
      name: this.state.name,
      price: parseFloat(this.state.price),
      date: this.state.date,
      category: this.state.category,
    };

    const budgetPlan = this.props.getBudgetPlan(item.date.substring(3, 11), item.category);
    budgetPlan.expenses += item.price;
    this.props.updateBudgetPlan(budgetPlan);
    this.props.addExpenses(item);
  }

  onUpdateItemHandler = () => {
    if (this.validate() === false) {
      return;
    }

    const item = {
      name: this.state.name,
      price: parseFloat(this.state.price),
      date: this.state.date,
      category: this.state.category,
      id: this.props.item.id,
    };

    this.props.updateExpenses(item);

    const hasChangeInPrice = this.props.item.price !== item.price;
    const hasChangeInCategory = this.props.item.category !== item.category;

    if (hasChangeInCategory || hasChangeInPrice) {
      const date = item.date.substring(3, 11);

      const currentBudgetPlan = this.props.getBudgetPlan(date, item.category);

      if (this.props.item.category !== item.category) {
        const previousBudgetPlan = this.props.getBudgetPlan(date, this.props.item.category);
        previousBudgetPlan.expenses -= this.props.item.price;
        this.props.updateBudgetPlan(previousBudgetPlan);

        currentBudgetPlan.expenses += item.price;
      } else {
        currentBudgetPlan.expenses += (item.price - this.props.item.price);
      }

      this.props.updateBudgetPlan(currentBudgetPlan);
    }
  }

  validate = () => {
    if (this.state.name === '' || this.state.price === '') {
      return false;
    }

    if (this.state.category === undefined || this.state.category === '') {
      return false;
    }

    if (Number.isNaN(parseFloat(this.state.price))) {
      return false;
    }

    return true;
  }

  addNewCategory = () => {
    this.props.navigator.showModal({
      screen: 'expenses.AddUpdateBudgetPlanModal',
      title: 'Add new category',
      animationType: 'slide-up',
      passProps: { date: this.state.date.substring(3, 11) },
    });
  }

  render() {
    let submitButton = <DefaultButton title="Add" onPress={this.onAddItemHandler} />;

    if (this.state.mode === 'edit') {
      submitButton = <DefaultButton title="Update" onPress={this.onUpdateItemHandler} />;
    }

    let categoryPicker = (
      <TouchableNativeFeedback onPress={this.addNewCategory}>
        <View >
          <Text style={styles.addNewCategoryWrapper}>Click to add new BudgetPlan</Text>
        </View>
      </TouchableNativeFeedback>
    );

    if (this.props.categories(this.state.date.substring(3, 11)).length > 0) {
      categoryPicker = (
        <DefaultPicker
          items={this.props.categories(this.state.date.substring(3, 11))}
          selectedValue={this.state.category}
          onValueChange={this.onCategoryChangeHandler}
        />
      );
    }

    return (
      <View style={styles.container}>
        <View style={styles.contentWrapper}>
          <View style={styles.fieldWrapper}>
            <Text style={styles.label}>Name</Text>
            <DefaultInput value={this.state.name} onChangeText={this.onNameChangeHandler} placeholder="Enter item name" />
          </View>
          <View style={styles.fieldWrapper}>
            <Text style={styles.label}>Price</Text>
            <DefaultInput keyboardType="numeric" value={this.state.price} onChangeText={this.onPriceChangeHandler} placeholder="Enter price" />
          </View>
          <View style={styles.fieldWrapper}>
            <Text style={styles.label}>Category</Text>
            {categoryPicker}
          </View>
          <View style={styles.fieldWrapper}>
            <Text style={styles.label}>Date</Text>
            <DefaultDatePicker
              date={this.state.date}
              format="DD-MMM-YYYY"
              confirmBtnText="Confirm"
              cancelBtnText="Cancel"
              onDateChange={this.onDateChangeHandler}
            />
          </View>
        </View>
        <View style={styles.addButtonWrapper}>
          {this.validate() ? submitButton : null}
        </View>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addExpenses: item => dispatch(addExpenses(item)),
  updateExpenses: item => dispatch(updateExpenses(item)),
  updateBudgetPlan: budgetPlan => dispatch(updateBudgetPlan(budgetPlan)),
});

const mapStateToProps = state => ({
  categories: date => _.map(
    _.filter(state.budgetPlan.budgetPlans, o => o.date === date),
    o => o.category,
  ),
  getBudgetPlan: (date, category) => _.find(state.budgetPlan.budgetPlans, { date, category }),
});

export default connect(mapStateToProps, mapDispatchToProps)(AddNewItemModal);
