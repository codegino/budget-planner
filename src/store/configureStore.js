import createSagaMiddleware from 'redux-saga';
import { createStore, combineReducers, applyMiddleware, compose } from 'redux';

import { watchAuth, watchExpenses, watchBudgetPlan } from './sagas/index';
import authReducer from './reducers/auth';
import uiReducer from './reducers/ui';
import expensesReducer from './reducers/expenses';
import budgetPlanReducer from './reducers/budgetPlan';

const sagaMiddleware = createSagaMiddleware();

const rootReducer = combineReducers({
  auth: authReducer,
  ui: uiReducer,
  expenses: expensesReducer,
  budgetPlan: budgetPlanReducer,
});

let composeEnhancers = compose;

if (__DEV__) {
  // eslint-disable-next-line
  composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const configureStore = createStore(rootReducer, composeEnhancers(applyMiddleware(sagaMiddleware)));

export default () => configureStore;

sagaMiddleware.run(watchAuth);
sagaMiddleware.run(watchExpenses);
sagaMiddleware.run(watchBudgetPlan);
