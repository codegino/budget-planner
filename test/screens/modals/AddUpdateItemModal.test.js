import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import AddUpdateItemModal from 'screens/modals/AddUpdateItemModal';

describe('Rendering of view', () => {
  const RealDate = Date;

  let store;
  beforeAll(() => {
    const constantDate = new Date('May 02 2018');
    // eslint-disable-next-line
    Date = class extends Date {
      constructor() {
        super();
        return constantDate;
      }
    };
  });

  afterAll(() => {
    // eslint-disable-next-line
    Date = RealDate;
  });

  beforeEach(() => {
    const getState = {
      budgetPlan: {
        budgetPlans: [],
      },
    };
    const mockStore = configureStore(getState);
    store = mockStore(getState);
    jest.mock('DatePickerIOS', () => 'DatePickerIOS');
  });

  it('will render the default view for adding new Item', () => {
    const wrapper = renderer.create(<AddUpdateItemModal store={store} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('will render the default view for edditing Item', () => {
    const item = {
      name: 'test name',
      category: 'test category',
      price: 100,
      date: 'May-02-2018',
    };

    const wrapper = renderer.create(<AddUpdateItemModal store={store} mode="edit" item={item} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
