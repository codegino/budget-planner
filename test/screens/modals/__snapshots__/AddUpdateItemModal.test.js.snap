// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering of view will render the default view for adding new Item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ddd",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingTop": 20,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Name
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter item name"
        placeholderTextColor="#5fbd98"
        style={
          Array [
            Object {
              "backgroundColor": "#193e2c",
              "borderColor": "#5fbd98",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 8,
              "marginTop": 8,
              "padding": 5,
              "paddingLeft": 10,
              "width": "100%",
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Price
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Enter price"
        placeholderTextColor="#5fbd98"
        style={
          Array [
            Object {
              "backgroundColor": "#193e2c",
              "borderColor": "#5fbd98",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 8,
              "marginTop": 8,
              "padding": 5,
              "paddingLeft": 10,
              "width": "100%",
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Category
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Date
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "width": 142,
            },
            Object {
              "backgroundColor": "#193e2c",
              "borderColor": "#3b9b75",
              "borderRadius": 10,
              "borderWidth": 1,
              "overflow": "hidden",
              "width": "100%",
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#193e2c",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#aaa",
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#193e2c",
                  "borderBottomLeftRadius": 9,
                  "borderColor": "#5fbd98",
                  "borderTopLeftRadius": 9,
                  "overflow": "hidden",
                },
                false,
                false,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
            >
              01-Jan-1970
            </Text>
          </View>
          <Image
            source={Object {}}
            style={
              Array [
                Object {
                  "height": 32,
                  "marginLeft": 5,
                  "marginRight": 5,
                  "width": 32,
                },
                undefined,
              ]
            }
          />
          <Modal
            animationType="none"
            hardwareAccelerated={false}
            onRequestClose={[Function]}
            supportedOrientations={
              Array [
                "portrait",
                "portrait-upside-down",
                "landscape",
                "landscape-left",
                "landscape-right",
              ]
            }
            transparent={true}
            visible={false}
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "flex-end",
                    "backgroundColor": "#00000077",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "height": 0,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View
                      pointerEvents="auto"
                    >
                      <DatePickerIOS
                        date={1970-01-01T00:00:00.000Z}
                        locale={undefined}
                        maximumDate={undefined}
                        minimumDate={undefined}
                        minuteInterval={undefined}
                        mode="date"
                        onDateChange={[Function]}
                        style={
                          Array [
                            Object {
                              "borderTopColor": "#ccc",
                              "borderTopWidth": 1,
                              "marginTop": 42,
                            },
                            undefined,
                          ]
                        }
                        timeZoneOffsetInMinutes={null}
                      />
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "height": 42,
                            "justifyContent": "center",
                            "paddingHorizontal": 20,
                            "position": "absolute",
                            "top": 0,
                          },
                          Object {
                            "left": 0,
                          },
                          undefined,
                        ]
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#46cf98",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#666",
                            },
                            undefined,
                          ]
                        }
                      >
                        Cancel
                      </Text>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "height": 42,
                            "justifyContent": "center",
                            "paddingHorizontal": 20,
                            "position": "absolute",
                            "top": 0,
                          },
                          Object {
                            "right": 0,
                          },
                          undefined,
                        ]
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#46cf98",
                              "fontSize": 16,
                            },
                            undefined,
                          ]
                        }
                      >
                        Confirm
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </Modal>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "margin": 20,
        "width": "80%",
      }
    }
  />
</View>
`;

exports[`Rendering of view will render the default view for edditing Item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ddd",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingTop": 20,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Name
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter item name"
        placeholderTextColor="#5fbd98"
        style={
          Array [
            Object {
              "backgroundColor": "#193e2c",
              "borderColor": "#5fbd98",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 8,
              "marginTop": 8,
              "padding": 5,
              "paddingLeft": 10,
              "width": "100%",
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value="test name"
      />
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Price
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Enter price"
        placeholderTextColor="#5fbd98"
        style={
          Array [
            Object {
              "backgroundColor": "#193e2c",
              "borderColor": "#5fbd98",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "white",
              "marginBottom": 8,
              "marginTop": 8,
              "padding": 5,
              "paddingLeft": 10,
              "width": "100%",
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value="100"
      />
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Category
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffbcbc",
              "borderColor": "red",
              "borderWidth": 1,
              "height": 100,
              "justifyContent": "center",
              "margin": 10,
              "width": 300,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#333333",
              "margin": 20,
            }
          }
        >
          TouchableNativeFeedback is not supported on this platform!
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Date
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "width": 142,
            },
            Object {
              "backgroundColor": "#193e2c",
              "borderColor": "#3b9b75",
              "borderRadius": 10,
              "borderWidth": 1,
              "overflow": "hidden",
              "width": "100%",
            },
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              },
              Object {
                "backgroundColor": "#193e2c",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "#aaa",
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                },
                Object {
                  "backgroundColor": "#193e2c",
                  "borderBottomLeftRadius": 9,
                  "borderColor": "#5fbd98",
                  "borderTopLeftRadius": 9,
                  "overflow": "hidden",
                },
                false,
                false,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
            >
              01-Jan-1970
            </Text>
          </View>
          <Image
            source={Object {}}
            style={
              Array [
                Object {
                  "height": 32,
                  "marginLeft": 5,
                  "marginRight": 5,
                  "width": 32,
                },
                undefined,
              ]
            }
          />
          <Modal
            animationType="none"
            hardwareAccelerated={false}
            onRequestClose={[Function]}
            supportedOrientations={
              Array [
                "portrait",
                "portrait-upside-down",
                "landscape",
                "landscape-left",
                "landscape-right",
              ]
            }
            transparent={true}
            visible={false}
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "flex-end",
                    "backgroundColor": "#00000077",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "height": 0,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View
                      pointerEvents="auto"
                    >
                      <DatePickerIOS
                        date={1970-01-01T00:00:00.000Z}
                        locale={undefined}
                        maximumDate={undefined}
                        minimumDate={undefined}
                        minuteInterval={undefined}
                        mode="date"
                        onDateChange={[Function]}
                        style={
                          Array [
                            Object {
                              "borderTopColor": "#ccc",
                              "borderTopWidth": 1,
                              "marginTop": 42,
                            },
                            undefined,
                          ]
                        }
                        timeZoneOffsetInMinutes={null}
                      />
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "height": 42,
                            "justifyContent": "center",
                            "paddingHorizontal": 20,
                            "position": "absolute",
                            "top": 0,
                          },
                          Object {
                            "left": 0,
                          },
                          undefined,
                        ]
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#46cf98",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#666",
                            },
                            undefined,
                          ]
                        }
                      >
                        Cancel
                      </Text>
                    </View>
                    <View
                      accessibilityComponentType={undefined}
                      accessibilityLabel={undefined}
                      accessibilityTraits={undefined}
                      accessible={true}
                      hasTVPreferredFocus={undefined}
                      hitSlop={undefined}
                      isTVSelectable={true}
                      nativeID={undefined}
                      onLayout={undefined}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "height": 42,
                            "justifyContent": "center",
                            "paddingHorizontal": 20,
                            "position": "absolute",
                            "top": 0,
                          },
                          Object {
                            "right": 0,
                          },
                          undefined,
                        ]
                      }
                      testID={undefined}
                      tvParallaxProperties={undefined}
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#46cf98",
                              "fontSize": 16,
                            },
                            undefined,
                          ]
                        }
                      >
                        Confirm
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </Modal>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "margin": 20,
        "width": "80%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#193e2c",
              "borderColor": "#5fbd98",
              "borderRadius": 20,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Update
        </Text>
      </View>
    </View>
  </View>
</View>
`;
